inmem.go

```diff
@@ -270,49 +275,34 @@ func (i *InmemSink) Data() []*IntervalMetrics {
 	return intervals
 }
 
-func (i *InmemSink) getExistingInterval(intv time.Time) *IntervalMetrics {
-	i.intervalLock.RLock()
-	defer i.intervalLock.RUnlock()
-
-	n := len(i.intervals)
-	if n > 0 && i.intervals[n-1].Interval == intv {
-		return i.intervals[n-1]
-	}
-	return nil
-}
-
-func (i *InmemSink) createInterval(intv time.Time) *IntervalMetrics {
+// getInterval returns the current interval. A new interval is created if no
+// previous interval exists, or if the current time is beyond the window for the
+// current interval.
+func (i *InmemSink) getInterval() *IntervalMetrics {
 	i.intervalLock.Lock()
 	defer i.intervalLock.Unlock()
```

Comment:
I guess there is a functional change here, where previously there was an optimization removing the need to get the write lock if the interval is still current. I'll restore that.